plugins/janus_sip.c

```diff
@@ -1383,6 +1397,28 @@ static void janus_sip_remove_quotes(char *str) {
 	}
 }
 
+static json_t *janus_sip_get_incoming_headers(const sip_t *sip, const janus_sip_session *session) {
+	json_t *headers = json_object();
+	if (!sip) {
+		return headers;
+	}
+        sip_unknown_t *unknown_header = sip->sip_unknown;
+        while (unknown_header != NULL) {
+                GList *temp = session->incoming_header_prefixes;
+                while (temp != NULL) {
+                        char *header_prefix = (char *) temp->data;
+                        if (strncmp(unknown_header->un_name, header_prefix, strlen(header_prefix)) == 0) {
```

Comment:
This should check if `header_prefix` and `unknown_header->un_name` are not NULL.